### PR TITLE
Fix make monitorinfo a struct

### LINKED ISSUE #1923: Make MONITORINFO a struct
This is possible fix for exception thrown when using CoreRT.

Feedback from CoreRT chatroom :

"Btw, you can work around the GetMonitorInfo trouble by making MONITORINFO a struct and passing it either as a pointer or a ref to the GetMonitorInfo method. The compiler can't generate marshaling for classes currently"

https://github.com/AvaloniaUI/Avalonia/blob/a0331b98896bffa09f02b0db999527fdf0fe2481/src/Windows/Avalonia.Win32/Interop/UnmanagedMethods.cs#L1058

Error when using WindowStartupLocation:
```
Unhandled Exception: System.Exception: Method '[Avalonia.Win32]Avalonia.Win32.Interop.UnmanagedMethods.GetMonitorInfo(native int,MONITORINFO)' requires non-trivial marshalling that is not yet supported by this compiler.
   at Avalonia.Win32.Interop.UnmanagedMethods.GetMonitorInfo(IntPtr, UnmanagedMethods.MONITORINFO) + 0x23
   at Avalonia.Win32.ScreenImpl.<>c__DisplayClass4_0.<get_AllScreens>b__0(IntPtr, IntPtr, Rect&, IntPtr) + 0x82
   at Core2D.Avalonia!<BaseAddress>+0x1475a4b
```

Error when using menus:
```
Unhandled Exception: System.Exception: Method '[Avalonia.Win32]Avalonia.Win32.Interop.UnmanagedMethods.GetMonitorInfo(native int,MONITORINFO)' requires non-trivial marshalling that is not yet supported by this compiler.
   at Avalonia.Win32.Interop.UnmanagedMethods.GetMonitorInfo(IntPtr, UnmanagedMethods.MONITORINFO) + 0x23
   at Avalonia.Win32.ScreenImpl.<>c__DisplayClass4_0.<get_AllScreens>b__0(IntPtr, IntPtr, Rect&, IntPtr) + 0x82
   at Core2D.Avalonia!<BaseAddress>+0x1475a4b
````

### PR DESCRIPTION
**Pull request by ahopper**
- What does the pull request do?
Passes MONITORINFO as a struct to overcome shortcomings in CoreRT
- What is the current behavior?
MONITORINFO is a class that CoreRT can't currently marshall
- What is the updated/expected behavior with this PR?
WindowsStartupLocation, menus and tooltips should work on windows CoreRT build

If the pull request fixes issue(s) list them like this:

Fixes #1923

### CODE REVIEW

File: src/Windows/Avalonia.Win32/Interop/UnmanagedMethods.cs [hunk 1]
```diff
@@ -1055,12 +1055,17 @@ public enum MONITOR
...
-            public int cbSize = Marshal.SizeOf<MONITORINFO>();
-            public RECT rcMonitor = new RECT();
-            public RECT rcWork = new RECT();
-            public int dwFlags = 0;
+            public int cbSize;
+            public RECT rcMonitor;
+            public RECT rcWork;
+            public int dwFlags;
+
+            public static MONITORINFO NewMONITORINFO()
```
Comment (grokys) [inline src/Windows/Avalonia.Win32/Interop/UnmanagedMethods.cs]: Nit: this would be better called `Create` I think.

**Review by grokys — CHANGES_REQUESTED** [review]
Looks good, just one tiny nit. @wieslawsoltes does this fix your problem?

File: src/Windows/Avalonia.Win32/Interop/UnmanagedMethods.cs (hunk 1 above)
Comment (ahopper) [reply src/Windows/Avalonia.Win32/Interop/UnmanagedMethods.cs]: Yep, makes sense to me,

File: src/Windows/Avalonia.Win32/Interop/UnmanagedMethods.cs [hunk 2]
```diff
@@ -1055,12 +1055,17 @@ public enum MONITOR
...
-            public int cbSize = Marshal.SizeOf<MONITORINFO>();
-            public RECT rcMonitor = new RECT();
-            public RECT rcWork = new RECT();
-            public int dwFlags = 0;
+            public int cbSize;
+            public RECT rcMonitor;
+            public RECT rcWork;
+            public int dwFlags;
+
+            public static MONITORINFO CreateMONITORINFO()
```
Comment (grokys) [inline src/Windows/Avalonia.Win32/Interop/UnmanagedMethods.cs]: I meant just `Create` ;) That's what static factory methods are usually called. Is there any reason it needs `MONITORINFO` in there?

**Review by grokys — APPROVED** [review]

**Comment by grokys** [comment]
Going to give @wieslawsoltes a little bit of time to respond, if he doesn't will merge tomorrow.

**Comment by wieslawsoltes** [comment]
@grokys please merge

**Pull request merged**